tracer/span.go

```diff
@@ -58,9 +58,9 @@ type Span struct {
 	Error    int32              `json:"error"`             // error status of the span; 0 means no errors
 	Sampled  bool               `json:"-"`                 // if this span is sampled (and should be kept/recorded) or not
 
-	tracer   *Tracer    // the tracer that generated this span
-	mu       sync.Mutex // lock the Span to make it thread-safe
-	finished bool       // true if the span has been submitted to a tracer.
+	sync.RWMutex
```

Comment:
Here we're slighly modifying our API and exposing the RWMutex methods AFAIK. Not a big big deal but will probably require another pair of eyes as this does change what we expose to users. I, for one, am fine with it.